content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md

```diff
@@ -1050,6 +1050,16 @@ The following is an example procedure to upgrade from `v1beta1` to `v1`.
 2.  Write an upgrade procedure to list all existing objects and write them with
     the same content. This forces the backend to write objects in the current
     storage version, which is `v1`.
-2. Remove `v1beta1` from the CustomResourceDefinition `status.storedVersions` field.
-
+3.  Remove `v1beta1` from the CustomResourceDefinition `status.storedVersions` field.
 
+{{< note >}}
+The `kubectl` tool currently cannot be used to edit or patch the `status` subresource on a CRD: see the [Kubectl Subresource Support KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2590-kubectl-subresource) for more details.
```

Comment:
Doing this via the CLI would be surprising to me... I would expect this to be done by the process upgrading the existing objects, so something like client-go could be used.

using `k8s.io/client-go/dynamic#Interface`, you could patch the status of a CRD like this:

```go
client, err := dynamic.NewForConfig(restClientConfig)
if err != nil {
  ... handle error ...
}

_, err = client.Resource(schema.GroupVersionResource{
  Group: "apiextensions.k8s.io",
  Version: "v1",
  Resource: "customresourcedefinitions",
}).Patch(
  context.TODO(),
  yourCRDName,
  types.MergePatchType,
  `{"status":{"storedVersions":[`+yourCRDCurrentStoredVersion+`]}}`,
  metav1.PatchOptions{},
  "status",
)
if err != nil {
  ... handle error ...
}
```

Reply:
Yes, of course doing it programmatically end eventually using an operator, is by far the best option.
But option 2 is about *manually* performing this and on step 1 I read *using kubectl* while it's absolutely not clear how to manually do the step 3.